### PR TITLE
JavaDoc for public API

### PR DESCRIPTION
**Pull request by d0rich**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -3,6 +3,21 @@ plugins {
     base
     // Kotlin plugin should be loaded only once
     id("com.gevamu.kotlin-common-conventions") apply false
+    // Generate api reference with dokka
+    id("org.jetbrains.dokka") version "1.7.20"
```
Comment (ASitnikov) [inline build.gradle.kts:7]: I think we need to use the same version as kotlin version

File: build.gradle.kts [hunk 2]
```diff
@@ -3,6 +3,21 @@ plugins {
...
+}
+
+// Apply dokka to subprojects and dokka dependencies
+allprojects {
```
Comment (ASitnikov) [inline build.gradle.kts:11]: Add dokka to kotlin conventions plugin

File: payments-contracts/src/main/kotlin/com/gevamu/corda/contracts/PaymentContract.kt [hunk 3]
```diff
@@ -20,6 +20,9 @@ import net.corda.core.contracts.CommandData
 import net.corda.core.contracts.Contract
 import net.corda.core.transactions.LedgerTransaction
 
+/**
```
Comment (ASitnikov) [inline payments-contracts/src/main/kotlin/com/gevamu/corda/contracts/PaymentContract.kt:23]: It's good to describe verification logic. Please speak with Andrey Senov and add it to the Pull Request which adds verification and on which Andrey's working on

File: payments-contracts/src/main/kotlin/com/gevamu/corda/states/Payment.kt [hunk 4]
```diff
@@ -29,22 +29,27 @@ import net.corda.core.serialization.CordaSerializable
 import java.time.Instant
 import java.util.UUID
 
+/**
+ * Data class describing Payment state in Corda business network
+ *
+ * @param payer Participant Corda node, which initiated payment
```
Comment (ASitnikov) [inline payments-contracts/src/main/kotlin/com/gevamu/corda/states/Payment.kt:35]: Here and after
Corda party, which ...

File: payments-contracts/src/main/kotlin/com/gevamu/corda/states/Payment.kt [hunk 5]
```diff
@@ -29,22 +29,27 @@ import net.corda.core.serialization.CordaSerializable
...
+ * @param gateway Gateway Corda node, which processes payment
+ * @param endToEndId Unique value for the payment in debtor-creditor scope
+ * @param paymentInstructionId Link to payment instruction, stored as attachment. See [AttachmentId]
+ * @param status Status of the payment in workflow
+ * @param additionalInfo Payment info provided by the bank.
+ *                      Null by default since initial payment doesn't have any bank response.
```
Comment (ASitnikov) [inline payments-contracts/src/main/kotlin/com/gevamu/corda/states/Payment.kt:41]: Null if there is no additional information associated with the state

File: payments-workflows/src/main/kotlin/com/gevamu/corda/flows/PaymentFlow.kt [hunk 6]
```diff
@@ -31,24 +31,64 @@ import net.corda.core.serialization.CordaSerializable
...
+ * @see Payment
+ */
 @StartableByRPC
 class PaymentFlow(
     private val paymentInstruction: PaymentInstruction,
     private val gateway: Party,
     private val uniquePaymentId: UUID = UUID.randomUUID(),
 ) : FlowLogic<List<StateAndRef<Payment>>>() {
 
+    /**
```
Comment (ASitnikov) [inline payments-workflows/src/main/kotlin/com/gevamu/corda/flows/PaymentFlow.kt:85]: I think comments on overriden methods can be omitted

**Review by ASitnikov — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-publish-sdk.yml [hunk 7]
```diff
@@ -28,6 +28,17 @@ jobs:
           cd workflows/build/libs/
           ls | xargs -I {} sh -c 'sha256sum "{}" > {}.sha256'
 
+      - name: 'Build Dokka API reference'
+        run: ./gradlew dokkaHtmlMultiModule
+
+      # Will push HTML files to gh-pages branch
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        if: github.ref == 'refs/heads/master'
```
Comment (alex-f-k) [inline .github/workflows/build-and-publish-sdk.yml:37]: This workflow already has a rule to run only on tag pushes. This line is redundant and should be removed

**Review by alex-f-k — CHANGES_REQUESTED** [review]
Please, change workflow yaml file

File: build.gradle.kts (hunk 1 above)
Comment (alex-f-k) [reply build.gradle.kts:7]: You can set manual version to the `0.7.5`. See https://mvnrepository.com/artifact/org.jetbrains.kotlinx/kotlinx-html-jvm/0.7.5

**Review by alex-f-k — DISMISSED** [review]

**Review by ASitnikov — APPROVED** [review]

**Review by alex-f-k — APPROVED** [review]

**Pull request merged**